tests/test_project/admin.py

```diff
@@ -24,9 +25,32 @@ class OperatorInline(admin.StackedInline):
     extra = 0
 
 
-class ProjectAdmin(admin.ModelAdmin):
+class ProjectAdmin(admin.ModelAdmin, UUIDAdmin):
     inlines = [OperatorInline]
     list_display = ['name']
+    readonly_fields = ['uuid']
+    fields = ['name', 'key', 'uuid']
+
+    class Media:
+        js = ('openwisp-utils/js/uuid.js',)
+
+    def _get_fields(self, fields, request, obj=None):
+        """
+        removes readonly_fields in add view
+        """
+        if obj:
+            return fields
+        new_fields = fields[:]
+        for field in self.readonly_fields:
+            if field in new_fields:
+                new_fields.remove(field)
+        return new_fields
```

Comment:
Can't this be generalized and added to the `UUIDAdmin` class?

Reply:
Correct, I think this should be generalized in the `UUIDAdmin` class.